pkg/scheduler/connections/pool.go

```diff
@@ -0,0 +1,155 @@
+package connections
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	schedulerv1pb "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+	"github.com/dapr/dapr/pkg/scheduler"
+	"github.com/dapr/kit/logger"
+)
+
+var log = logger.NewLogger("dapr.runtime.scheduler")
+
+// Pool represents a connection pool for namespace/appID separation of sidecars to schedulers.
+type Pool struct {
+	Lock             sync.RWMutex
+	NsAppIDPool      map[string]*AppIDPool
+	MaxConnsPerAppID int // future expand to diff conn count for diff appIDs?
+}
+
+// AppIDPool represents a pool of connections for a single appID.
+type AppIDPool struct {
+	lock        sync.RWMutex
+	connections []*Connection
+}
+
+type Connection struct {
+	ConnDetails *scheduler.SidecarConnDetails
+	Stream      schedulerv1pb.Scheduler_WatchJobServer
+}
+
+// Add adds a connection to the pool for a given namespace/appID.
+func (p *Pool) Add(nsAppID string, conn *Connection) {
+	p.Lock.Lock()
+	defer p.Lock.Unlock()
+
+	// Ensure the AppIDPool exists for the given nsAppID
+	if p.NsAppIDPool[nsAppID] == nil {
+		p.NsAppIDPool[nsAppID] = &AppIDPool{
+			lock:        sync.RWMutex{},
+			connections: make([]*Connection, 0),
+		}
+	}
+
+	// Check if adding the connection would exceed the maximum connection count
+	if len(p.NsAppIDPool[nsAppID].connections) >= p.MaxConnsPerAppID {
+		log.Infof("Sufficient number of Sidecar connections to Scheduler reached. Not adding connection for namespace/appID: %s. Current connection count: %d\n", nsAppID, len(p.NsAppIDPool[nsAppID].connections))
```

Comment:
Why? This number will only ever be arbitrary and is a nob which no one will be able to tune correctly, especially when it ""should"" change during runtime. Please remove entirely. Applications expect to load balance jobs across an entire replica set.